### PR TITLE
diagnostic: refactor git origin checks and also check Cask origin

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

c.f. https://discourse.brew.sh/t/issue-installing-recently-added-cask/4206/

This pull request refactors the git origin checks in `brew doctor` to use a common helper function, and also checks the Cask tap, if it exists, for a suspicious origin URL.

No tests written because the test framework doesn't use an actual git folder for its temporary repository.

### CODE REVIEW

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -106,6 +106,34 @@ def please_create_pull_requests(what = "unsupported configuration")
...
+            Without a correctly configured origin, Homebrew won't update
+            properly. You can solve this by adding the remote:
+              git -C "#{repository_path}" remote add origin #{Formatter.url("https://github.com/#{desired_origin}.git")}
+          EOS
+        elsif current_origin !~ %r{#{desired_origin}(\.git|/)?$}i
+          <<~EOS
+            Suspicious #{desired_origin} git origin remote found.
+
+            With a non-standard origin, Homebrew won't pull updates from
+            the main repository. The current git origin is:
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb]: Should replace `the main repository` with something else that makes sense

File: Library/Homebrew/diagnostic.rb [hunk 2]
```diff
@@ -106,6 +106,34 @@ def please_create_pull_requests(what = "unsupported configuration")
...
+              #{current_origin}
+
+            Unless you have compelling reasons, consider setting the
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb]: Remove `Unless you have compelling reasons`; people should consider it regardless and their compelling reasons can overrule them 😆

**Comment by MikeMcQuaid** [comment]
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming) to the rescue!

**Comment by jonchang** [comment]
Updated the wording to mirror each message. Will merge when green.

**Comment by MikeMcQuaid** [comment]
Thanks @jonchang!

**Pull request merged**